.github/workflows/docs.yaml

```diff
@@ -0,0 +1,134 @@
+name: Docs
+
+on:
+  push:
+    branches:
+    - 'master'
+    - 'v[0-9]+.[0-9]+'
+    paths:
+    - 'docs/**'
+    - '.github/workflows/docs.yaml'
+  pull_request:
+    branches:
+    - '**'
+    paths:
+    - 'docs/**'
+    - '.github/workflows/docs.yaml'
+    types:
+    - opened
+    - edited
+    - reopened
+    - synchronize
+  schedule:
+  - cron:  '0 23 * * *' # daily at 11pm
+
+env:
+  retention_days: 90
+
+jobs:
+  build-and-test:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Set up Poetry
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        ./docs/hack/install-poetry.sh
+
+    - name: Install dependencies
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        pushd ./docs/
+        ~/.poetry/bin/poetry install
+
+    - name: Build docs
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+        
+        make -C ./docs/ multiversion SPHINXOPTS="--keep-going"
+
+        if [[ ! -d ./docs/_build ]]; then
+          echo "docs dir not found"
+          exit 1
+        fi
+        
+        tar -cJf ~/docs.tar.xz -C ./docs/_build .
+
+    - name: Upload docs artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: docs.tar.xz
+        path: ~/docs.tar.xz
+        if-no-files-found: error
+        retention-days: ${{ env.retention_days }}
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' && github.event_name == 'push' }}
```

Comment:
I won't publish backports. We either need to deploy also on pushes to release branches, or enable manual button with deploy step

Reply:
yeah, I need to redo it. This checking out random tips of other branches it a nightmare for the CI.